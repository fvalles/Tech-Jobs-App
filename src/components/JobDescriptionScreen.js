/* eslint-disable no-undef */
/* eslint-disable no-alert */
import React from 'react';
import { ScrollView } from 'react-native';
import PropTypes from 'prop-types';
import Markdown from 'react-native-markdown-display';
import { StyledText } from './StyledText';
import { StyledTouchableOpacity } from './StyledTouchableOpacity';
import { StyledView } from './StyledView';
import { storeData } from './AsyncStorageHelper';

export default function JobDescriptionScreen({ route }) {
  const { companyName, jobDesc, jobId, jobTitle } = route.params;
  return (
    <ScrollView>
      <StyledView viewType="jobDesc">
        <Markdown>{jobDesc}</Markdown>
      </StyledView>
      <StyledView viewType="flexSaveBtn">
        <StyledTouchableOpacity
          touchableType="descriptionBtn"
          onPress={() => {
            storeData({ id: jobId, companyName, jobDesc, jobTitle });
            alert('Job Saved to Favorites!');
          }}
        >
          <StyledText textType="descBtn">Save Job</StyledText>
        </StyledTouchableOpacity>
      </StyledView>
    </ScrollView>
  );
}

JobDescriptionScreen.propTypes = {
  route: PropTypes.shape({
    params: PropTypes.shape({
      companyName: PropTypes.string.isRequired,
      jobDesc: PropTypes.string.isRequired,
      jobId: PropTypes.string.isRequired,
      jobTitle: PropTypes.string.isRequired,
    }),
  }).isRequired,
};
