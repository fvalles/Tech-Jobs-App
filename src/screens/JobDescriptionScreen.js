/* eslint-disable no-undef */
/* eslint-disable no-alert */
import React from 'react';
import { ScrollView } from 'react-native';
import PropTypes from 'prop-types';
import Markdown from 'react-native-markdown-display';
import { StyledText } from '../components/StyledText';
import { StyledTouchableOpacity } from '../components/StyledTouchableOpacity';
import { StyledView } from '../components/StyledView';
import { storeData } from '../utils/asyncStorageHelper';

export default function JobDescriptionScreen({ route }) {
  const { companyName, isFavJob, jobDesc, jobId, jobTitle } = route.params;
  return (
    <ScrollView>
      <StyledView viewType="jobDesc">
        <Markdown>{jobDesc}</Markdown>
      </StyledView>
      {!isFavJob && (
        <StyledView viewType="flexSaveBtn">
          <StyledTouchableOpacity
            touchableType="descriptionBtn"
            onPress={() => {
              storeData({ id: jobId, companyName, jobDesc, jobTitle });
              alert('Job Saved to Favorites!');
            }}
          >
            <StyledText textType="descBtn">Save Job</StyledText>
          </StyledTouchableOpacity>
        </StyledView>
      )}
    </ScrollView>
  );
}

JobDescriptionScreen.propTypes = {
  route: PropTypes.shape({
    params: PropTypes.shape({
      companyName: PropTypes.string.isRequired,
      isFavJob: PropTypes.bool,
      jobDesc: PropTypes.string.isRequired,
      jobId: PropTypes.string.isRequired,
      jobTitle: PropTypes.string.isRequired,
    }),
  }).isRequired,
};
