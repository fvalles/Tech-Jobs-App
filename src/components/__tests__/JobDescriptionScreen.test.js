import React from 'react';
import renderer from 'react-test-renderer';

import JobDescriptionScreen from '../JobDescriptionScreen';

describe('<JobDescriptionScreen />', () => {
  const route = { params: { jobDesc: 'Mock Job Description' } };
  it('has 1 child', () => {
    const tree = renderer.create(<JobDescriptionScreen route={route} />).toJSON();
    expect(tree.children.length).toBe(1);
  });
  it('renders correctly', () => {
    const tree = renderer.create(<JobDescriptionScreen route={route} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
