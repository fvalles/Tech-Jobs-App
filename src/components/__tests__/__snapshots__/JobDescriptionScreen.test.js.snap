// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<JobDescriptionScreen /> renders correctly 1`] = `
<RCTScrollView>
  <View>
    <View
      style={
        Array [
          Object {
            "paddingBottom": 5,
            "paddingLeft": 15,
            "paddingRight": 15,
            "paddingTop": 5,
          },
        ]
      }
      viewType="jobDesc"
    >
      <View
        style={Object {}}
      >
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "flexDirection": "row",
              "flexWrap": "wrap",
              "justifyContent": "flex-start",
              "marginBottom": 10,
              "marginTop": 10,
              "width": "100%",
            }
          }
        >
          <Text
            style={Object {}}
          >
            <Text
              style={
                Array [
                  Object {},
                  Object {},
                ]
              }
            >
              Mock Job Description
            </Text>
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
